### ISSUE TITLE
`/metrics` and `/api_metrics` endpoint does not show the generic API metrics for the director's endpoints

### ISSUE THREAD

**Issue by Malsourie** [cloudfoundry/bosh#2467]
**Describe the bug**
According to [documentation](https://bosh.io/docs/director-metrics/#metrics-server) the `/metrics` and `/api_metrics` endpoint will expose the generic API metrics for the director's endpoints including number of requests and response time. But when we call those endpoints:
1. `/metrics` only exposes the metrics for `/metrics` and `/api_metrics` endpoints
1. `/api_metrics` only return OK

**To Reproduce**

1. Deploy a bosh director 
1. curl some endpoints of bosh, e.g. `vms`, `deployments`, etc.
1. curl http://<bosh_ip>:9092/metrics
1. curl curl http://<bosh_ip>:9092/api_metrics

**Expected behavior**
The endpoint should return metrics of bosh endpoints.

**Comment by mvach** [comment]
Hi,
as show case I implemented a tiny puma and prometheus-client integration which serves the expected webserver access metrics https://github.com/mvach/PumaMetricsExample.

Sadly I don't see the difference to the current director implementation right now.

**Comment by beyhan** [comment]
I'm not sure how the generic API metrics are supposed to work at all because:

- As defined in the [director job](https://github.com/cloudfoundry/bosh/blob/3a31bc2a63d69a7bc1df8d7bac2a5cc8d6ca2b71/jobs/director/templates/metrics-server#L5-L6) the metrics-server is started as a different process. You can check this also on a director VM with `netstat -tulpn | grep 9092` & `ps -aux | grep <pid-from-previous-command>`.
- [bosh-director-metrics-server](https://github.com/cloudfoundry/bosh/blob/3a31bc2a63d69a7bc1df8d7bac2a5cc8d6ca2b71/src/bosh-director/bin/bosh-director-metrics-server#L28-L40) starts and registers the Prometheus collector for itself
- the [metrics_collector](https://github.com/cloudfoundry/bosh/blob/main/src/bosh-director/lib/bosh/director/metrics_collector.rb) collects only the bosh metrics and no generic API metrics.

You see only metrics for the `/metrics` endpoint because this is the only endpoint you call on the metrics server. Maybe I miss something here but this is my current understanding.

**Comment by mvach** [comment]
:-) @beyhan,
I just noticed that right now and wanted to update the issue.

**Comment by beyhan** [comment]
Good catch @jpalermo. [This](https://github.com/cloudfoundry/bosh/commit/f0873641303095818b34d39035ee0a672d305e77) is the commit which introduced the change. It looks like initially the `/api_metrics` were called `/director_metrics` but yes they redirect to the director process which I missed. 

>The transition from thin to puma might have introduced separate datasets for each of the forked processes puma creates. So it does return data, but I didn't have a chance yet to verify that the data is actually correct.

This is a good question. It looks to me that they should be accurate because the director internal metrics are gathered from the DB data, but you "never know" :-)
